### PR TITLE
Disk performance dashboard

### LINKED ISSUE #1304: [2] Create dashboard dedicated for disk metrics
**Is your feature request related to a problem? Please describe.**
Since we want to use local disk volumes for performance then we should also monitor this performance.

**Describe the solution you'd like**
Dashboard dedicated for disk metrics that can highlight local disk volumes.

**Definition of done:**
Dedicated dashboard added.

### PR DESCRIPTION
**Pull request by Zash**
> [!warning]
> **This is public repository, ensure not to disclose:**
> - [ ] personal data beyond what is necessary for interacting with this pull request
> - [ ] business confidential information, such as customer names

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [x] kind/feature       
- [ ] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [ ] kind/admin-change  
- [ ] kind/dev-change    
- [ ] kind/security      
- [ ] [kind/adr]()       

### What does this PR do / why do we need this PR?

Adds a dashboard dedicated to disk and filesystem performance.

- Fixes #1304

#### Additional information to reviewers

Changes from "Node Exporter Full":
- Removed all rows except Disk and Filesystem related ones
- Allows selecting multiple hosts
- Selectable disk devices and filesystem mount points
- New Overview row

#### Screenshots

image

#### Checklist

- [x] Proper commit message prefix on all commits
  
- Change checks:
  - [x] The change is transparent
  - [ ] The change is disruptive
  - [ ] The change requires no migration steps
  - [ ] The change requires migration steps
- Metrics checks:
  - [ ] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [ ] The logs do not show any errors after the change
- Network Policy checks:
  - [ ] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packages in the `NetworkPolicy Dashboard`
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy

### CODE REVIEW

**Comment by robinAwallace** [comment]
Looks nice :+1: Just one thought, the "Disk space used" board. Is it used or how much it is left. The title says used but the colors says how much is left :smile: Either flip the colors or to disk space left?

**Comment by Zash** [comment]
> the "Disk space used" board. Is it used or how much it is left

Hah, in my attempt to simplify the query I seem to have inverted it, it used to be `100-(avail * 100 / total) --> 0-100`  and now it's `avail/total --> 0.0-1.0` (and then Grafana turns that into %).  Will fix.

**Comment by Zash** [comment]
This should make more sense

screenshot of disk usage graph showing high percentages

File: helmfile/charts/grafana-dashboards/dashboards/node-exporter-disk-dashboard.json [hunk 1]
```diff
@@ -0,0 +1,2745 @@
...
+                "color": "dark-green",
+                "value": null
+              },
+              {
+                "color": "green",
+                "value": 70
+              },
+              {
+                "color": "yellow",
+                "value": 80
```
Comment (robinAwallace) [inline helmfile/charts/grafana-dashboards/dashboards/node-exporter-disk-dashboard.json:133]: I would probably lower this to 70% or 65%
I think that would look better :thinking:

Comment (Zash) [reply helmfile/charts/grafana-dashboards/dashboards/node-exporter-disk-dashboard.json]: [How about](https://github.com/elastisys/compliantkubernetes-apps/pull/1796/commits/45fd5c8b49b48c400ed70d354d9d4338bf6e21c1)
- 🟩 <55%
- 🟨 ~70%
- 🟥 >85%

image

Comment (robinAwallace) [reply helmfile/charts/grafana-dashboards/dashboards/node-exporter-disk-dashboard.json]: Yeah I think that looks better

File: helmfile/charts/grafana-dashboards/dashboards/node-exporter-disk-dashboard.json [hunk 2]
```diff
@@ -0,0 +1,2753 @@
...
+          "fields": "",
+          "values": false
+        },
+        "showUnfilled": true,
+        "valueMode": "color"
+      },
+      "pluginVersion": "9.5.5",
+      "targets": [
+        {
+          "expr": "1-(node_filesystem_avail_bytes{instance=~\"$node\", job=\"$job\", mountpoint=~\"$mountpoint\"} / node_filesystem_size_bytes{instance=~\"$node\", job=\"$job\", mountpoint=~\"$mountpoint\"})",
```
[the remark this answers is not in the document]
Comment (Zash) [reply helmfile/charts/grafana-dashboards/dashboards/node-exporter-disk-dashboard.json:171]: Oh it does but only on the other term of that division. Maybe that's what caused some NaN issues.

**Review by Xartos — APPROVED** [review]

**Review by robinAwallace — APPROVED** [review]

**Pull request merged**